### PR TITLE
Make it possible to override the UID/GID/username of the user running

### PR DESCRIPTION
**Pull request by andriisoldatenko**
resolves https://github.com/astronomer/airflow-operator/issues/1161




Example:
```
docker build --build-arg UID=1001 --build-arg GID=1001 -t tt .
[+] Building 1.3s (9/9) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                       0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                                                                                                                            0.0s
 => [internal] load metadata for docker.io/library/redis:6.2.7                                                                                                                                                                                             0.0s
 => [1/5] FROM docker.io/library/redis:6.2.7                                                                                                                                                                                                               0.0s
 => CACHED [2/5] RUN apt-get update &&     apt-get dist-upgrade -y &&     rm -rf /var/lib/apt/lists/*                                                                                                                                                      0.0s
 => [3/5] RUN groupmod -g 1001 redis   && chgrp 1001 /data                                                                                                                                                                                                 0.2s
 => [4/5] RUN usermod -u 1001 -g 1001 redis                                                                                                                                                                                                                0.4s
 => [5/5] RUN chown -R 1001:1001 /usr/local/bin/redis*                                                                                                                                                                                                     0.4s
 => exporting to image                                                                                                                                                                                                                                     0.1s
 => => exporting layers                                                                                                                                                                                                                                    0.1s
 => => writing image sha256:809ae6cfe235507b6268168e0a9191ccca407d64e618d2097cc32fe8dec622f9                                                                                                                                                               0.0s
 => => naming to docker.io/library/tt                                                                              
```

```
$ docker run -it tt sh
$ id
uid=1001(redis) gid=1001(redis) groups=1001(redis)
$
```

### CODE REVIEW

**Review by ashb — APPROVED** [review]

File: redis/Dockerfile [hunk 1]
```diff
@@ -23,12 +23,25 @@ RUN apt-get update && \
...
-RUN chown redis:redis /usr/local/bin/redis*
+RUN chown -R $UID:$GID /usr/local/bin/redis*
 
 ARG BUILD_NUMBER=-1
 LABEL io.astronomer.docker=true
 LABEL io.astronomer.docker.build.number=$BUILD_NUMBER
 
 # Switching user to redis:999
-USER redis
+USER $UNAME
```
Comment (andriisoldatenko) [inline redis/Dockerfile:47]:
```suggestion
USER $UID:$GID
```
@pgvishnuram do you mean this?

Comment (pgvishnuram) [reply redis/Dockerfile:47]: ```USER $UNAME -> USER $UID ```

File: redis/Dockerfile [hunk 2]
```diff
@@ -23,12 +23,25 @@ RUN apt-get update && \
     apt-get dist-upgrade -y && \
     rm -rf /var/lib/apt/lists/*
 
+# Make it possible to override the UID/GID/username of the user running
+ARG UID=999
+ARG GID=999
+
+# Since username and group name `redis` already exists in base image
+ENV UNAME=redis
```
Comment (danielhoherd) [inline redis/Dockerfile:31]: Given that `uname` is something different, and `$USER` is already used to hold the username, I think we should do this:

```suggestion
ENV USER=redis
```

Comment (danielhoherd) [reply redis/Dockerfile:31]: Or do we even need this ENV var to hold the username, since it will always be "redis"?

**Comment by andriisoldatenko** [comment]
@danielhoherd PTAL

**Review by danielhoherd — APPROVED** [review]
I still really dislike the variable `UNAME` being used to indicate the user name, and think we should use `USER` or `USERNAME`. Other than that, this all looks good.

EDIT: just to add some reasoning, `ENV` vars in Dockerfiles persist to the runtime environment. a `UNAME` var may be misinterpreted by something that expects that to contain `uname` style info. Really low chance of this happening, but it's better to err on the side of caution when it's so easy to select a more canonical variable name for storing the user name.

**Comment by andriisoldatenko** [comment]
> I still really dislike the variable `UNAME` being used to indicate the user name, and think we should use `USER` or `USERNAME`. Other than that, this all looks good.

yeah i can fix it, one moment.

**Review by danielhoherd — APPROVED** [review]
LGTM

**Review by danielhoherd — CHANGES_REQUESTED** [review]
@andriisoldatenko We'll need to bump the version.txt in all of these dirs to successfully build and ship the new versions to the docker repos.

**Comment by andriisoldatenko** [comment]
> @andriisoldatenko We'll need to bump the version.txt in all of these dirs to successfully build and ship the new versions to the docker repos.

@danielhoherd done!.

File: redis/version.txt [hunk 3]
```diff
@@ -1 +1 @@
-6.2.7
+6.2.8
```
Comment (danielhoherd) [inline redis/version.txt:1]: Because we're still using `FROM redis:6.2.7`:
```suggestion
6.2.7-1
```

**Review by danielhoherd — APPROVED** [review]

**Review by danielhoherd — APPROVED** [review]

**Review by danielhoherd — APPROVED** [review]

**Pull request merged**